core-authentication/src/test/java/gg/solarmc/loader/authentication/test/AuthenticationCenterIT.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * DataLoader
+ * Copyright © 2021 SolarMC Developers
+ *
+ * DataLoader is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * DataLoader is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with DataLoader. If not, see <https://www.gnu.org/licenses/>
+ * and navigate to version 3 of the GNU Affero General Public License.
+ */
+
+package gg.solarmc.loader.authentication.test;
+
+import gg.solarmc.loader.DataCenter;
+import gg.solarmc.loader.OnlineSolarPlayer;
+import gg.solarmc.loader.authentication.AuthenticationCenter;
+import gg.solarmc.loader.authentication.AutoLoginResult;
+import gg.solarmc.loader.authentication.CompleteLoginResult;
+import gg.solarmc.loader.authentication.CreateAccountResult;
+import gg.solarmc.loader.authentication.LoginAttempt;
+import gg.solarmc.loader.authentication.User;
+import gg.solarmc.loader.authentication.VerifiablePassword;
+import gg.solarmc.loader.authentication.internal.UUIDOperations;
+import gg.solarmc.loader.impl.SolarDataConfig;
+import gg.solarmc.loader.impl.UserDetails;
+import gg.solarmc.loader.impl.test.extension.DataCenterInfo;
+import gg.solarmc.loader.impl.test.extension.DatabaseExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(DatabaseExtension.class)
+@ExtendWith(MockitoExtension.class)
+public class AuthenticationCenterIT {
```

Comment:
This is a good example of a test for future reference when writing other tests that have to do with databases :)